quantum/debounce/eager_pk.c

```diff
@@ -44,6 +44,17 @@ static bool                matrix_need_update;
 #define DEBOUNCE_ELAPSED 251
 #define MAX_DEBOUNCE (DEBOUNCE_ELAPSED - 1)
 
+static uint16_t custom_wrap_timer_reference = 0;
+static uint8_t custom_wrap_timer_last_value = 0;
+
+static uint8_t custom_wrap_timer_read(void) {
+    uint16_t custom_wrap_timer_new_reference = timer_read();
+    uint16_t diff = custom_wrap_timer_new_reference - custom_wrap_timer_reference;
+    custom_wrap_timer_reference = custom_wrap_timer_new_reference;
+    custom_wrap_timer_last_value = (custom_wrap_timer_last_value + diff) % (MAX_DEBOUNCE + 1);
+    return custom_wrap_timer_last_value;
+}
```

Comment:
The `TIMER_DIFF` change LGTM.

As for `custom_wrap_timer_read()`, I'm not a fan of the naming scheme. Shorter variable names would make the code easier to understand at a glance. Also, the globals can be made static local since they aren't used outside of the function body (this also precludes name collisions). How about this?

```suggestion
static uint8_t wrap_timer_read(void) {
    static uint16_t time = 0;
    static uint8_t  last_result = 0;

    uint16_t new_time = timer_read();
    uint16_t diff = new_time - time;
    time = new_time;
    last_result = (last_result + diff) % (MAX_DEBOUNCE + 1);
    return last_result;
}
```

Reply:
Alright, changes committed